WalletWasabi.Fluent/Behaviors/AutoHideFlyoutBehavior.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System.Linq;
+using System.Reactive.Disposables;
+using System.Reactive.Linq;
+using Avalonia.Controls;
+using Avalonia.Controls.Primitives;
+using Avalonia.VisualTree;
+using ReactiveUI;
+
+namespace WalletWasabi.Fluent.Behaviors;
+
+/// <summary>
+/// The behavior intend to fix https://github.com/zkSNACKs/WalletWasabi/issues/10181.
+/// </summary>
+public class AutoHideFlyoutBehavior : DisposingBehavior<Window>
+{
+	protected override void OnAttached(CompositeDisposable disposables)
+	{
+		AssociatedObject?
+			.WhenAnyValue(x => x.IsActive, x => x.IsPointerOver, (isActive, isPointerOver) => !isActive && !isPointerOver)
+			.Where(x => x)
+			.Subscribe(_ => HideAllFlyout(AssociatedObject))
+			.DisposeWith(disposables);
+	}
+
+	private void HideAllFlyout(Control parent)
+	{
+		FlyoutBase.GetAttachedFlyout(parent)?.Hide();
+
+		foreach (var child in parent.GetVisualChildren().OfType<Control>())
```

Comment:
Quite expensive. As an improvement, you can listen for flyout open events in the behavior and remember them if they belong to elements under the associated object. Closing a flyout removes it from a list of opened flyous. Then instead of that expensive method you can iterate over that list and call `Hide` for each of them.